### PR TITLE
New SLI and refactorting for scalability

### PR DESCRIPTION
**Pull request by pacospace**
## This introduces a breaking change

- [x] Yes
- [ ] No

## This Pull Request implements

- Refactored main file
- Created main report class and references for the report
- Introduced base class for all metrics to ease the creation of new ones.
- Introduced HTML templates using Jinja2
- Added SLI for Thoth Knowledge Graph
- Added SLI for Thoth Learning
- Added SLI for Thoth User-API
- Added SLI for adviser, solver, inspection workflows

### CODE REVIEW

**Comment by goern** [comment]
```
[WARNING][08:00:54] PycodestyleBear (E501): This result has no patch attached.

thoth/slo_reporter/sli_solver_quality.py
|  54| ············"avg_solver_workflows_failed":·f"avg_over_time(thoth_workflows_status{query_labels_solver_workflows_f}[{_INTERVAL}])",
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (130 > 120 characters)

thoth/slo_reporter/sli_solver_quality.py
|  55| ············"avg_solver_workflows_error":·f"avg_over_time(thoth_workflows_status{query_labels_solver_workflows_e}[{_INTERVAL}])",
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (129 > 120 characters)

thoth/slo_reporter/sli_solver_quality.py
|  78| ························Thoth·Solver·was·successfull·<strong>{int(successfull_percentage)}%·</strong>·of·the·time·in·the·last·week.·\
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (133 > 120 characters)
[WARNING][08:00:55] PycodestyleBear (E501): This result has no patch attached.
[WARNING][08:00:55] PycodestyleBear (E501): This result has no patch attached.
[WARNING][08:00:55] PycodestyleBear (E501): This result has no patch attached.
[WARNING][08:00:55] PycodestyleBear (E501): This result has no patch attached.
[WARNING][08:00:55] PycodestyleBear (E501): This result has no patch attached.

thoth/slo_reporter/sli_adviser_quality.py
|  47| ········query_labels_adviser_workflows_f·=·f'\''{{instance="{_INSTANCE}",·label_selector="component=adviser",·job="Thoth·Metrics·({_ENVIRONMENT})",·workflow_status="Failed"}}'\''
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (172 > 120 characters)

thoth/slo_reporter/sli_adviser_quality.py
|  48| ········query_labels_adviser_workflows_e·=·f'\''{{instance="{_INSTANCE}",·label_selector="component=adviser",·job="Thoth·Metrics·({_ENVIRONMENT})",·workflow_status="Error"}}'\''
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (171 > 120 characters)

thoth/slo_reporter/sli_adviser_quality.py
|  52| ············"avg_adviser_workflows_failed":·f"avg_over_time(thoth_workflows_status{query_labels_adviser_workflows_f}[{_INTERVAL}])",
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (132 > 120 characters)

thoth/slo_reporter/sli_adviser_quality.py
|  53| ············"avg_adviser_workflows_error":·f"avg_over_time(thoth_workflows_status{query_labels_adviser_workflows_e}[{_INTERVAL}])",
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (131 > 120 characters)

thoth/slo_reporter/sli_adviser_quality.py
|  76| ························Thoth·Adviser·was·successfull·<strong>{int(successfull_percentage)}%·</strong>·of·the·time·in·the·last·week.·\
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (134 > 120 characters)
[WARNING][08:00:55] PycodestyleBear (E501): This result has no patch attached.
[WARNING][08:00:55] PycodestyleBear (E501): This result has no patch attached.

thoth/slo_reporter/sli_learning_rate.py
|  48| ············"max_learning_rate":·f"max_over_time(increase(thoth_graphdb_unsolved_python_package_versions_change_total{query_labels}[1h])[{_INTERVAL}:1h])"
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (154 > 120 characters)

thoth/slo_reporter/sli_learning_rate.py
|  57| ····················Thoth·Learning·rate·reached·<strong>{int(sli['\''max_learning_rate'\''])}·packages/hour·</strong>·in·the·last·week.·\
|    | [NORMAL] PycodestyleBear (E501):
|    | E501 line too long (131 > 120 characters)
```

File: thoth/slo_reporter/templates/pypi_knowledge_graph.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{% include "base.html" %}
+{% block title %} <strong> PyPI Knowledge Graph </strong> {% endblock %}
+{% block classes %}
+<thead>
+    <tr>
+        <th>Parameter</th>
```
Comment (goern) [inline thoth/slo_reporter/templates/pypi_knowledge_graph.html:6]: do we need this? it is confusing the user/reader

Comment (pacospace) [reply thoth/slo_reporter/templates/pypi_knowledge_graph.html:6]: Didn't we want to show differences in terms of global knowledge?

Comment (pacospace) [reply thoth/slo_reporter/templates/pypi_knowledge_graph.html:6]: It's not activated yet in any case, as metrics are not yet scraped for that.

**Pull request merged**